hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/BaseFileOnlyRelation.scala

```diff
@@ -36,26 +36,29 @@ import org.apache.spark.sql.types.StructType
  * [[BaseRelation]] implementation only reading Base files of Hudi tables, essentially supporting following querying
  * modes:
  * <ul>
- * <li>For COW tables: Snapshot</li>
- * <li>For MOR tables: Read-optimized</li>
+ *  <li>For COW tables: Snapshot</li>
+ *  <li>For MOR tables: Read-optimized</li>
  * </ul>
  *
- * NOTE: The reason this Relation is used in liue of Spark's default [[HadoopFsRelation]] is primarily due to the
+ * NOTE: The reason this Relation is used in-liue of Spark's default [[HadoopFsRelation]] is primarily due to the
  * fact that it injects real partition's path as the value of the partition field, which Hudi ultimately persists
  * as part of the record payload. In some cases, however, partition path might not necessarily be equal to the
  * verbatim value of the partition path field (when custom [[KeyGenerator]] is used) therefore leading to incorrect
  * partition field values being written
  */
-class BaseFileOnlyRelation(sqlContext: SQLContext,
-                           metaClient: HoodieTableMetaClient,
-                           optParams: Map[String, String],
-                           userSchema: Option[StructType],
-                           globPaths: Seq[Path])
-  extends HoodieBaseRelation(sqlContext, metaClient, optParams, userSchema) with SparkAdapterSupport {
+case class BaseFileOnlyRelation(override val sqlContext: SQLContext,
```

Comment:
Primary change here is converting the class to be a case class, which in turn entails that all of the ctor parameters would become field values requiring corresponding annotation

Reply:
The reason this method is converted to a case-class is to avoid any in-place mutations and instead make `updatePrunedDataSchema` produce new instance instead